src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionEmitMemberAccessor.cs

```diff
@@ -10,6 +10,11 @@
 
 namespace System.Text.Json.Serialization.Metadata
 {
+#if NETCOREAPP
+    [RequiresDynamicCode("Runtime code generation is not available for Aot form factor")]
+    [UnconditionalSuppressMessage("AotAnalysis", "IL3051:RequiresDynamicCode",
```

Comment:
Same - we shouldn't need this. If this type is RDC, we can't construct an instance of it without a warning. It should be okay to have mismatched virtual override annotations after that because we'd be warned at the time the instance is constructed.